### PR TITLE
Allow enable pod priority feature gate for GCE and configure priority for kube-proxy

### PR DESCRIPTION
**Pull request by MrHohn**
**What this PR does / why we need it**:
From #23225, this PR adds an option for user to enable pod priority feature gate using GCE startup scripts, and configure pod priority for kube-proxy when enabled.

The setup `priorityClassName: system` derives from: https://github.com/kubernetes/kubernetes/blob/ce1485c626d8cddc9dbffe24f91f7d62480148cd/staging/src/k8s.io/api/core/v1/types.go#L2536-L2542

The plan is to configure pod priority for kube-proxy daemonset (https://github.com/kubernetes/kubernetes/pull/50705) in the same way.

**Special notes for your reviewer**:
cc @bsalamat @davidopp @thockin 

**Release note**:

```release-note
When using kube-up.sh on GCE, user could set env `ENABLE_POD_PRIORITY=true` to enable pod priority feature gate.
```

### CODE REVIEW

File: cluster/gce/container-linux/configure-helper.sh [hunk 1]
```diff
@@ -692,11 +692,16 @@ function start-kube-proxy {
   if [[ -n "${ENABLE_CACHE_MUTATION_DETECTOR:-}" ]]; then
     container_env="env:\n    - name: KUBE_CACHE_MUTATION_DETECTOR\n    value: \"${ENABLE_CACHE_MUTATION_DETECTOR}\""
   fi
+  local pod_priority=""
+  if [[ "${ENABLE_POD_PRIORITY}" == "true" ]]; then
+    pod_priority="priorityClassName: system"
```
Comment (bsalamat) [inline cluster/gce/container-linux/configure-helper.sh]: s/system/system-cluster-critical
ref: https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/admission/priority/admission.go#L47

File: cluster/saltbase/salt/kube-proxy/kube-proxy.manifest [hunk 2]
```diff
@@ -31,6 +31,11 @@
 
 {% set throttles = "--iptables-sync-period=1m --iptables-min-sync-period=10s" -%}
 
+{% set pod_priority = "" -%}
+{% if pillar.get('enable_pod_priority', '').lower() == 'true' -%}
+  {% set pod_priority = "priorityClassName: system" -%}
```
Comment (bsalamat) [inline cluster/saltbase/salt/kube-proxy/kube-proxy.manifest]: s/system/system-cluster-critical

File: cluster/gce/gci/configure-helper.sh [hunk 3]
```diff
@@ -1030,11 +1030,16 @@ function start-kube-proxy {
   if [[ -n "${ENABLE_CACHE_MUTATION_DETECTOR:-}" ]]; then
     container_env="env:\n    - name: KUBE_CACHE_MUTATION_DETECTOR\n    value: \"${ENABLE_CACHE_MUTATION_DETECTOR}\""
   fi
+  local pod_priority=""
+  if [[ "${ENABLE_POD_PRIORITY}" == "true" ]]; then
+    pod_priority="priorityClassName: system"
```
Comment (bsalamat) [inline cluster/gce/gci/configure-helper.sh]: s/system/system-cluster-critical

File: cluster/gce/container-linux/configure-helper.sh (hunk 1 above)
Comment (MrHohn) [inline cluster/gce/container-linux/configure-helper.sh]: Thanks :) Done.

File: cluster/saltbase/salt/kube-proxy/kube-proxy.manifest (hunk 2 above)
Comment (MrHohn) [inline cluster/saltbase/salt/kube-proxy/kube-proxy.manifest]: Done.

File: cluster/gce/gci/configure-helper.sh (hunk 3 above)
Comment (MrHohn) [inline cluster/gce/gci/configure-helper.sh]: Done.

**Comment by bsalamat** [comment]
Sorry! I made a mistake in my suggested priority class. Those `system-cluster-critical` priority classes should be changed to `system-node-critical`, because kube-proxy is supposed to run on all  the nodes. system-cluster-critical is more appropriate for critical system pods which need to exist in every cluster, but they don't need to be present on every node.

**Comment by MrHohn** [comment]
@bsalamat Thanks for clarifying, fixed now.

**Comment by bsalamat** [comment]
Thanks, @MrHohn!
Setting the right priority class name for Kube-proxy LGTM. I would like someone more familiar with the scripts to review this PR as well.

**Comment by MrHohn** [comment]
/retest

cc @kubernetes/sig-node-pr-reviews for setting priority on kube-proxy.

**Comment by MrHohn** [comment]
> PR needs rebase

Rebased.

**Comment by MrHohn** [comment]
Rebased. Changes now include setting pod priority on kube-proxy daemonset as well.

@kubernetes/sig-cluster-lifecycle-pr-reviews

**Comment by MrHohn** [comment]
Ping, would anyone has bandwidth take a look? I believe it wouldn't take that long :)

cc @mikedanese @dchen1107

**Review by dnardo — COMMENTED** [review]
/lgtm

**Comment by MrHohn** [comment]
/assign @dchen1107 
for approval.

/unassign @jszczepkowski

**Comment by luxas** [comment]
Is Pod Priority an alpha or beta feature in v1.8?
Should kubeadm do this as well?

**Comment by MrHohn** [comment]
> Is Pod Priority an alpha or beta feature in v1.8?

@luxas I believe Pod Priority is alpha in v1.8 (https://github.com/kubernetes/features/issues/268), what about wait till it hit beta?

**Comment by luxas** [comment]
> @luxas I believe Pod Priority is alpha in v1.8 (kubernetes/features#268), what about wait till it hit beta?

@MrHohn that also sounds good to me. Was just curious if something had changes as you added support for it here (even behind a feature flag though).
Thanks!

**Pull request merged**